programming/sr/vision/index.md

```diff
@@ -171,62 +184,71 @@ One (possibly both) of them may change to resolve this.
 
 An `Orientation` object describes the orientation of a marker.
 
-![A visual representation of how the orientation axes work. Source: SourceBots]({{ site.baseurl }}/images/content/vision/yaw-pitch-roll.png)
-
-pitch
+rot_x
 :   Rotation of the marker about the cartesian x-axis, in radians.
 
-    Leaning a marker away from the camera increases the value of `rot_x`, while
-    leaning it towards the camera decreases it. A value of 0 indicates that the
-    marker is upright.
+    Leaning a marker towards the camera increases the value of `rot_x`, while
+    leaning it away from the camera decreases it. A value of π or -π indicates
+    that the marker is upright.
 
-yaw
+rot_y
 :   Rotation of the marker about the cartesian y-axis, in radians.
 
-    Turning a marker clockwise (as viewed from above) increases the value of
-    `rot_y`, while turning it anticlockwise decreases it. A value of 0 means
+    Turning a marker clockwise (as viewed from above) decreases the value of
+    `rot_y`, while turning it anticlockwise increases it. A value of 0 means
     that the marker is perpendicular to the line of sight of the camera.
 
-roll
+rot_z
 :   Rotation of the marker about the cartesian z-axis, in radians.
 
     Turning a marker anticlockwise (as viewed from the camera) increases the
-    value of `rot_z`, while turning it clockwise decreases it. A value of 0
-    indicates that the marker is upright.
+    value of `rot_z`, while turning it clockwise decreases it. A value of π
+    or -π indicates that the marker is upright.
 
-rot_x
-:   An alias for `pitch`.
+roll
+:   An alias for `rot_x`.
 
-rot_y
-:   An alias for `yaw`.
+pitch
+:   An alias for `rot_y`.
```

Comment:
```suggestion
:   A rotation about the transverse axis, in radians (An axis right to left across the token). Tilting the marker backward increases the value of `pitch`,
    while tilting it forwards decreases it. A value of 0 indicates that the marker is facing the camera square-on.
    
    This differs from `rot_x` in the zero point and direction to increase the value.
```

Reply:
This implies that it is conceptually more or different than `rot_y`, which it is not. This property is merely an alias for the same quantity.

Reply:
`pitch` is __not__ an alias of `rot_y` unless you are using april_vision, in zoloto mode the axis and zero-points of yaw, pitch and roll differ from the axis rotations.

Reply:
Both `Orientation.rot_y` and `Orientation.pitch` are implemented as `return self._yaw_pitch_roll[1]`, with no accounting for the mode of operation. They are thus identical. While the implementation may not be that one is strictly an alias of the other, that is functionally what they are.

Perhaps you're thinking of the spherical coordinates?

Reply:
`Orientation.rot_y` is implemented as `return self.yaw_pitch_roll[1]`, `Orientation.pitch` is implemented as `return self._yaw_pitch_roll[1]` (note the leading underscore). `_yaw_pitch_roll` is built from the quaternion that has not had the zoloto rotation applied in the init.

Reply:
Eugh, I didn't spot that. That's remarkably unclear code!

Will update this section with the details of the yaw/pitch/roll properties.

Reply:
6de4868